helix-term/src/ui/completion.rs

```diff
@@ -154,8 +154,19 @@ impl Completion {
                     );
                     doc.apply(&transaction, view.id);
 
-                    if let Some(additional_edits) = &item.additional_text_edits {
-                        // gopls uses this to add extra imports
+                    // apply additional edits, mostly used to auto import unqualified types
+                    let resolved_additional_text_edits = if item.additional_text_edits.is_some() {
+                        None
+                    } else {
+                        Completion::resolve_completion_item(doc, item.clone())
+                            .and_then(|item| item.additional_text_edits)
+                    };
+
+                    if let Some(additional_edits) = item
+                        .additional_text_edits
+                        .as_ref()
+                        .or_else(|| resolved_additional_text_edits.as_ref())
```

Comment:
Why not calculate `resolved_additional_text_edits` in `or_else` branch, that way we can just directly do roughly like
```suggestion
                    if let Some(additional_edits) = item
                        .additional_text_edits
                        .as_ref()
                        .or_else(||
                             // apply additional edits, mostly used to auto import unqualified types
                             Completion::resolve_completion_item(doc, item.clone())
                            .and_then(|item| item.additional_text_edits))
```

Reply:
Well, I also tried this. With that:
```rust
```

Reply:
And I cannot use "or_else" before without using "as_ref", because the compiler gives me the error:
"cannot move out of `item.additional_text_edits` which is behind a shared reference"

Reply:
I'll mark this a resolved, but fell free to open it again if you have any other suggestion